### DISCUSSION TITLE
Control which sprites are being painted first.

### DISCUSSION THREAD

**Discussion by The-DevBlog** [bevyengine/bevy#5094] [Q&A] [answered]
I am building a 2d top-down game. I have a moveable player and a map (a png file). The issue is that the map is being rendered on top of the player. How can I control which system is being run first in that frame? I'm away of the *system order of execution* from the *Bevy Cheat Book*, but I was not able to resolve the issue with that guide (nor am I sure that is what I should of been looking at). 

Thanks for any help!

Here is a simplified version of my code. 
```rust 
fn main() {
    App::new()
        .add_startup_system(build_map)
        .add_startup_system(spawn_player);
}

fn build_map(mut commands: Commands, assets: Res<AssetServer>) {
    let map = SpriteBundle {
        sprite: Sprite {
            custom_size: Some(Vec2::new(1920.0, 1080.0)),
            ..default()
        },
        texture: assets.load("map.png"),
        ..default()
    };

    commands.spawn_bundle(map);
}

// spawns the player to the game window on load
fn spawn_player(mut commands: Commands) {
    let sprite_bundle = SpriteBundle {
        sprite: Sprite {
            color: Color::MIDNIGHT_BLUE,
            custom_size: Some(Vec2::new(50.0, 50.0)),
            ..default()
        },
        ..default()
    };

    commands
        .spawn_bundle(sprite_bundle)
        .insert(Player)
        .insert(Speed(150.0));
}

#[derive(Component)]
struct Player;

#[derive(Component)]
struct Speed(f32);
```

This is the result. As you can see, the **player** is being rendered behind the map on each frame, so the player is never visible. 
<img width="955" alt="image" src="https://user-images.githubusercontent.com/52691521/175751798-c0ceb64a-0a8d-4650-a4fd-23b8ab65e7ff.png">

**Comment by alice-i-cecile** [comment, marked as answer]
As a first step, have you set the z `Transform` coordinate of all of your sprites correctly? This is the current way to control the rendering order of 2D sprites, although #1275 discusses some alternative ideas for how this might be done in the future.

**Reply by The-DevBlog** [reply]
Wow that was an easy fix 😅 Thank you Alice!
